mount-daemon/controllers/clientmount_controller.go

```diff
@@ -237,26 +235,24 @@ func (r *ClientMountReconciler) mount(ctx context.Context, clientMountInfo dwsv1
 
 	device, err := r.getDevice(clientMountInfo)
 	if err != nil {
-		return err
+		return dwsv1alpha2.NewResourceError("could not get device to mount").WithError(err)
 	}
 
 	// Create the mount file or directory
 	switch clientMountInfo.TargetType {
 	case "directory":
 		if err := r.mkdir(clientMountInfo.MountPath); err != nil {
-			log.Error(err, "Could not create mount directory", "mount path", clientMountInfo.MountPath, "device", device)
-			return err
+			return dwsv1alpha2.NewResourceError("could not make mount directory: '%s' device: '%s'", clientMountInfo.MountPath, device).WithError(err).WithMajor()
 		}
 	case "file":
 		// Create the parent directory and then the file
 		if err := r.mkdir(filepath.Dir(clientMountInfo.MountPath)); err != nil {
-			log.Error(err, "Could not create mount parent directory", "mount path", clientMountInfo.MountPath, "device", device)
-			return err
+
+			return dwsv1alpha2.NewResourceError("could not make parent mount directory: '%s' device: '%s'", clientMountInfo.MountPath, device).WithError(err).WithMajor()
```

Comment:
Should directory and device variables show up in the structured error output instead?

Reply:
Looks like there are several examples of this below too.